data/basics/userBalance.go

```diff
@@ -111,15 +111,33 @@ type OnlineAccountData struct {
 	VotingData
 }
 
+// GenesisAccountData contains a subset of account information that is
+// present in the genesis file.
+type GenesisAccountData struct {
+	_struct struct{} `codec:",omitempty,omitemptyarray"`
+
+	Status     Status     `codec:"onl"`
+	MicroAlgos MicroAlgos `codec:"algo"`
+
+	VoteID       crypto.OneTimeSignatureVerifier `codec:"vote"`
+	SelectionID  crypto.VRFVerifier              `codec:"sel"`
+	StateProofID merklesignature.Commitment      `codec:"stprf"`
+
+	VoteFirstValid  Round  `codec:"voteFst"`
+	VoteLastValid   Round  `codec:"voteLst"`
+	VoteKeyDilution uint64 `codec:"voteKD"`
+}
+
 // AccountData contains the data associated with a given address.
 //
 // This includes the account balance, cryptographic public keys,
 // consensus delegation status, asset data, and application data.
 type AccountData struct {
 	_struct struct{} `codec:",omitempty,omitemptyarray"`
 
-	Status     Status     `codec:"onl"`
-	MicroAlgos MicroAlgos `codec:"algo"`
+	// GenesisAccountData contains a subset of account information that is
```

Comment:
This looks good. This comports with the goal of getting the [SDK's Account type](https://github.com/algorand/go-algorand-sdk/blob/a935089a4bb5e07ca223a17f28114e179be94243/types/genesis.go#L59) to have an equivalent in `go-algorand`. Currently the **cross repo types check** [is failing for bookkeeping.Gensis vs. types.Genesis](https://github.com/algorand/go-algorand-sdk/blob/a935089a4bb5e07ca223a17f28114e179be94243/types/genesis.go#L59) because this very `AccountData` has too many fields. I believe that the test will still fail because of the path `Genesis.Allocation --> GenesisAllocation.Astate == types.Account` but maybe a different test can be carved out.

Reply:
That's a good point, we are thinking of changing the `State` struct to point to this in a follow up PR. 

I tried doing that at one point in this PR but reverted back because it got too big and I couldn’t isolate failing test changes: [https://github.com/algorand/go-algorand/pull/5447/commits/0a3e4f67c10c270beb21c06fc7ad146a811f6e80#diff-66b1b333be13b86893[…]419332b0574dad5c7b8R162](https://github.com/algorand/go-algorand/pull/5447/commits/0a3e4f67c10c270beb21c06fc7ad146a811f6e80#diff-66b1b333be13b868934a358b2b79b07c7c8dfd24bf27b419332b0574dad5c7b8R162)

Reply:
I agree that it's worth re-investigating with the changes smaller.